src/irmin-pack/inode.ml

```diff
@@ -345,25 +360,53 @@ struct
 
       let of_target : type ptr. ptr layout -> ptr t -> ptr = function
         | Total -> fun target -> Total_ptr target
-        | Partial _ ->
-            fun target -> { target = Some target; target_hash = target.hash }
+        | Partial _ -> fun target -> { target = Newie target }
         | Truncated -> fun target -> Intact target
 
       let of_hash : type ptr. ptr layout -> hash -> ptr = function
         | Total -> assert false
-        | Partial _ -> fun hash -> { target = None; target_hash = lazy hash }
+        | Partial _ -> fun hash -> { target = Lazy hash }
         | Truncated -> fun hash -> Broken hash
 
-      let iter_if_loaded :
+      let save :
```

Comment:
The sole purpose of this function is to be used from `Val.save`. Since it now also does stuff depending on the `clear` parameter, it is even more tied to `Val.save`. I don't see a better naming.

Reply:
Sure, but it doesn't actually _do_ the `save`: it's an iterator intended for use by a function that _does_ save the values being iterated over. OTOH, we have precedent for this sort of naming (e.g. `Tree.Contents.export` doesn't actually do an export), so we can keep this as-is.

Reply:
I was clearly influenced by `Tree.Contents.export` 🙄 

If you have a suggestion please share. I'm out of idea.

Reply:
I still like `iter_if_loaded`, but I don't strongly dislike `save` so I retract my comment :-) Apologies for the bikeshedding.

Reply:
No worries